sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignature.java

```diff
@@ -306,6 +321,15 @@ public static TimerParameter timerParameter(TimerDeclaration decl) {
     }
 
     /**
+     * Descriptor for a {@link Parameter} of a subtype of {@link PipelineOptions}.
+     */
+    @AutoValue
+    public abstract static class PipelineOptionsParameter extends Parameter {
+      PipelineOptionsParameter() {}
+      public abstract TypeDescriptor<? extends PipelineOptions> pipelineOptionsT();
```

Comment:
Is this necessary - can we just provide a parameter of type exactly `PipelineOptions`? It's trivial for the user to convert it to the needed type with `.as()`, and in some cases the user may need to `.as()` it to several different classes, which makes the current implementation overkill.